docs/design/build-system.md

```diff
@@ -0,0 +1,162 @@
+# Build System
+
+Angular CLI includes a first-party build system for Angular applications distributed as `@angular-devkit/build-angular`.
+This build system is responsible for creating a standalone application from user source files and third-party dependencies.
+
+`@angular-devkit/build-angular` itself integrates with the rest of Angular CLI by being an [Architect builder](https://angular.io/guide/cli-builder).
+This document describes a top level view of the functionality in `@angular-devkit/build-angular`, referred as just the "build system".
+Deprecated or soon to be removed features are not described here.
+
+In broad strokes the main areas are:
+- loading and processing sources
+- code splitting
+- production optimizations
+- post-processing steps
+
+Many tools are used in this process, and most of these steps happen within a [Webpack](https://webpack.js.org/) build.
+We maintain a number of Webpack-centric plugins in this repository, some of these are public but most are private since they are very specific to our setup.
+
+
+## Loading and processing sources
```

Comment:
It might be useful to have a diagram showing the different steps and how they relate to Webpack module loading/plugins etc.